staging/src/k8s.io/apiserver/plugin/pkg/authorizer/webhook/webhook.go

```diff
@@ -256,6 +286,36 @@ func (w *WebhookAuthorizer) RulesFor(user user.Info, namespace string) ([]author
 	return resourceRules, nonResourceRules, incomplete, fmt.Errorf("webhook authorizer does not support user rule resolution")
 }
 
+// Match is used to evaluate the SubjectAccessReviewSpec against
+// the authorizer's matchConditions in the form of cel expressions
+// to return match or no match found, which then is used to
+// determine if the webhook should be skipped.
+func (w *WebhookAuthorizer) match(ctx context.Context, r *authorizationv1.SubjectAccessReview) (bool, error) {
+	// A nil celMapper or zero saved CompilationResults matches all requests.
+	if w.celMapper == nil || w.celMapper.CompilationResults == nil {
+		return true, nil
+	}
+	evalResults, err := w.celMapper.Eval(ctx, r)
+	if err != nil {
+		return false, fmt.Errorf("authz: %w", err)
+	}
+	// If at least one matchCondition evaluates to FALSE, then the webhook is skipped.
+	// If ALL matchConditions evaluate to TRUE, then the webhook is called.
+	for _, evalResult := range evalResults {
+		if evalResult.EvalResult.Type() != celgo.BoolType {
+			return false, fmt.Errorf("authz: error evaluating SubjectAccessReview expression, should evaluate to bool but got %v", evalResult.EvalResult.Type())
```

Comment:
returning an error here doesn't match the comments... we should accumulate errors and return them at the end if no match conditions returned `false` successfully

update the tests to exercise this scenario